### PR TITLE
chore: add IpVersion guard to HashIpv6DifferentAddresses case

### PR DESCRIPTION
**Pull request by spacewander**
Commit Message: add IpVersion guard to HashIpv6DifferentAddresses case
Additional Description:
Risk Level: Low
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: test/common/router/config_impl_test.cc [hunk 1]
```diff
@@ -3166,6 +3166,12 @@ TEST_F(RouterMatcherHashPolicyTest, HashIpv4DifferentAddresses) {
 }
 
 TEST_F(RouterMatcherHashPolicyTest, HashIpv6DifferentAddresses) {
+  if (!TestEnvironment::shouldRunTestForIpVersion(Network::Address::IpVersion::v6)) {
+    // an exception like "IPv6 addresses are not supported on this machine" will be thrown
+    // if we don't add IP version check and run test on a machine that doesn't support IPv6
+    return;
```
Comment (soulxu) [inline test/common/router/config_impl_test.cc:3172]: we should use `GTEST_SKIP` here, something like `GTEST_SKIP() << the machine aren't support IPv6`

**Review by soulxu — CHANGES_REQUESTED** [review]

File: test/common/router/config_impl_test.cc (hunk 1 above)
Comment (spacewander) [reply test/common/router/config_impl_test.cc]: @soulxu 
Thanks. Update according to the review. The CI failure seems to be caused by failing to upload the docker image: https://dev.azure.com/cncf/4684fb3d-0389-4e0b-8251-221942316e06/_apis/build/builds/166826/logs/492

**Review by soulxu — APPROVED** [review]
LGTM, thanks!

**Pull request merged**